File: Arrays/ContainerWithMostWater.c
#include <stdio.h>


int maxArea(int* heightArray, int heightArraySize) {
    int maxContainerArea = 0;


    int inwardMovingPointer = 0;


    int outwardMovingPointer = heightArraySize - 1;


    while (outwardMovingPointer != inwardMovingPointer) {
        int minPointerYCoordinates = heightArray[inwardMovingPointer] < heightArray[outwardMovingPointer] ? 
                        heightArray[inwardMovingPointer] : 
                        heightArray[outwardMovingPointer];


        int currentContainerArea = (outwardMovingPointer - inwardMovingPointer) * minPointerYCoordinates;


        if (currentContainerArea > maxContainerArea) {
            maxContainerArea = currentContainerArea;
        }


        if (heightArray[outwardMovingPointer] > heightArray[inwardMovingPointer]) {
            inwardMovingPointer++;
        } 
        else {
            outwardMovingPointer--;
        }
    }


    return maxContainerArea;
}

File: Arrays/ContainerWithMostWater.js
var maxArea = function(heightArray) {
    let maxContainerArea = 0;


    let inwardMovingPointer = 0;


    let outwardMovingPointer = heightArray.length - 1;


    while (outwardMovingPointer !== inwardMovingPointer) {
        const minPointerYCoordinates = Math.min(heightArray[inwardMovingPointer], heightArray[outwardMovingPointer]);


        const currentContainerArea = (outwardMovingPointer - inwardMovingPointer) * minPointerYCoordinates;


        maxContainerArea = Math.max(maxContainerArea, currentContainerArea);


        if (heightArray[outwardMovingPointer] > heightArray[inwardMovingPointer]) {
            inwardMovingPointer++;
        } else {
            outwardMovingPointer--;
        }
    }


    return maxContainerArea;
};
